pkg/controller/podautoscaler/horizontal.go

```diff
@@ -652,7 +652,9 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 		var metricTimestamp time.Time
 		metricDesiredReplicas, metricName, metricStatuses, metricTimestamp, err = a.computeReplicasForMetrics(hpa, scale, hpa.Spec.Metrics)
 		if err != nil {
-			a.setCurrentReplicasInStatus(hpa, currentReplicas)
+			// set desiredReplicas to scale.Spec.Replicas when can't compute metrics
+			desiredReplicas = scale.Spec.Replicas
+			a.setStatus(hpa, currentReplicas, desiredReplicas, hpa.Status.CurrentMetrics,false)
```

Comment:
Nit: Add a space between `CurrentMetrics,` and `false`.  Running `gofmt` should fix this.

Reply:
yeah,i fixed it and added an unit test.

Reply:
@josephburnett